cmd/wasm-client/main.go

```diff
@@ -288,12 +288,24 @@ func (w *wasmClient) IsConnected(_ js.Value, _ []js.Value) interface{} {
 	return js.ValueOf(w.lndConn != nil)
 }
 
-func (w *wasmClient) Disconnect(_ js.Value, _ []js.Value) interface{} {
+func (w *wasmClient) Disconnect(_ js.Value, args []js.Value) interface{} {
 	if w.lndConn != nil {
-		if err := w.lndConn.Close(); err != nil {
-			log.Errorf("Error closing RPC connection: %v", err)
-		}
-		w.lndConn = nil
+		// We launch the closure of the connection in a goroutine to avoid
+		// that the JS websocket freezes and blocks while closing.
+		go func() {
+			if err := w.lndConn.Close(); err != nil {
+				log.Errorf("Error closing RPC connection: %v", err)
+			}
+			w.lndConn = nil
```

Comment:
Thanks for the review suggestion @ellemouton!

Unfortunately I did try that before, but using a `WaitGroup` and then `Wait`ing for the `WaitGroup` at the end of the `Disconnect` function function makes the same problem that existed prior to the fix occur. The main thread then blocks on the `Wait` and for some reason it never picks up that the `Websocket` is now Hard closed by the new updates in the `grpc` dependency. I think this is because this is in a JavaScript environment, and the main thread therefore never picks up any changes to the Javascript objects as it's blocking. So the `WaitGroup` `Wait` would never complete.

I therefore chose the alternative to signal the completion by a callback instead, as that unblocks the main thread, and makes it pick up the hard close of the Websocket. Hope that makes sense!

Reply:
makes sense! thanks for the explanation :)